### PR TITLE
compiletest: print the correct basename of the src dir

### LINKED ISSUE #40712: x.py test: compile-fail-fulldeps failures are reported as if they are in compile-fail
When I naively run `x.py test --stage 1`, I will get failures from tests that require a stage 2 compiler. This is expected behavior.

However, I also expect for the report of the failure to reflect the fact that the failing tests were in one of the subdirectories dedicated to such tests, such as `compile-fail-fulldeps` or `run-pass-fulldeps`.

Unfortunately, today it seems that if you do the naive run of `x.py test --stage 1`, the summary of the failing tests presents a very misleading picture:

```
failures:
    [compile-fail] compile-fail/proc-macro/derive-bad.rs
    [compile-fail] compile-fail/proc-macro/derive-still-gated.rs
    [compile-fail] compile-fail/proc-macro/expand-to-unstable-2.rs
    [compile-fail] compile-fail/proc-macro/expand-to-unstable.rs
    [compile-fail] compile-fail/proc-macro/import.rs
    [compile-fail] compile-fail/proc-macro/issue-37788.rs
    [compile-fail] compile-fail/proc-macro/issue-38586.rs
    [compile-fail] compile-fail/proc-macro/item-error.rs
    [compile-fail] compile-fail/proc-macro/load-panic.rs
    [compile-fail] compile-fail/proc-macro/no-macro-use-attr.rs
    [compile-fail] compile-fail/proc-macro/proc-macro-attributes.rs
    [compile-fail] compile-fail/proc-macro/resolve-error.rs
    [compile-fail] compile-fail/proc-macro/shadow.rs

test result: FAILED. 28 passed; 13 failed; 8 ignored; 0 measured
```

I can live with the fact that its using the prefix `[compile-fail]`; I don't know why it is, but if people want a short category there, fine.

But it *really* should also present the correct file path after that prefix (e.g. compile-fail-fulldeps/proc-macro/derive-bad.rs), both so that
 * I get my reminder that I need to be more discerning when testing a stage 1 compiler, and
 * Someone trying to find the test actually knows where to look for it.

### PR DESCRIPTION
**Pull request by RalfJung**
See <https://github.com/laumann/compiletest-rs/issues/76>.

Fixes #40712

### CODE REVIEW

**Comment by kennytm** [comment]
Does this close #40712?

File: src/tools/compiletest/src/main.rs [hunk 1]
```diff
@@ -517,7 +517,7 @@ pub fn make_test_name(config: &Config, testpaths: &TestPaths) -> test::TestName
     //
     //    run-pass/foo/bar/baz.rs
     let path =
-        PathBuf::from(config.mode.to_string())
+        PathBuf::from(config.src_base.file_name().unwrap());
```
Comment (laumann) [inline src/tools/compiletest/src/main.rs]: `config.src_base` is already a `PathBuf` so wouldn't it suffice to just write `config.src_base`?

**Comment by RalfJung** [comment]
@kennytm Yes I think it does.

File: src/tools/compiletest/src/main.rs (hunk 1 above)
Comment (RalfJung) [reply src/tools/compiletest/src/main.rs]: That will print the absolute path to the test. I wanted to make it relative to the current working directory, and this is the crude approximation I ended up with.

Comment (laumann) [reply src/tools/compiletest/src/main.rs]: Ah OK

**Pull request merged**